libraries/opensk/src/api/fingerprint.rs

```diff
@@ -56,16 +54,28 @@ pub trait Fingerprint {
     /// `prepare_enrollment` must be called first.
     /// A returned `Ctap2StatusCode` indicates an unexpected failure processing
     /// the command.
-    /// The `Ctap2EnrollFeedback` contains expected errors from the fingerprint
-    /// capture process.
-    /// Also returns the expected number of remaining samples.
+    ///
+    /// This function returns:
+    /// - The `Ctap2EnrollFeedback` contains expected errors from the
+    /// fingerprint capture process.
+    /// - The expected number of remaining samples.
+    /// If the sensor finished template creation, return a hardware ID for this
+    /// template. This ID is different from the FIDO template ID, but they are
+    /// mapped to each other in OpenSK. This should happen exactly when the
+    /// returned number of `remainingSamples` is 0.
+    ///
+    /// There is always only one fingerprint enrollment in progress, and the
+    /// capture implicitly refers to that.
     fn capture_sample(
         &mut self,
-        template_id: &[u8],
         timeout_ms: Option<usize>,
-    ) -> CtapResult<(Ctap2EnrollFeedback, usize)>;
+    ) -> CtapResult<(Ctap2EnrollFeedback, usize, Option<Vec<u8>>)>;
 
     /// Cancel a fingerprint enrollment.
+    ///
+    /// Returns the newly assigned hardware template ID. This is not to be
```

Comment:
copy paste? This whole paragraph seems out of context.

Reply:
Indeed, removed.